kotlin/wipe_appwrite_collection/src/Index.kt

```diff
@@ -0,0 +1,110 @@
+import com.google.gson.Gson
+import io.appwrite.Client
+import io.appwrite.services.Databases
+import io.appwrite.models.DocumentList
+
+/*
+  'req' variable has:
+    'headers' - object with request headers
+    'payload' - request body data as a string
+    'variables' - object with function variables
+  'res' variable has:
+    'send(text, status)' - function to return text response. Status code defaults to 200
+    'json(obj, status)' - function to return JSON response. Status code defaults to 200
+  
+  If an error is thrown, a response with code 500 will be returned.
+*/
+
+@Throws(Exception::class)
+suspend fun main(req: RuntimeRequest, res: RuntimeResponse): RuntimeResponse {
+    val HTTP_BAD_REQEST = 400
+
+    // Check for API endpoint and key in the variables
+    if (req.variables["APPWRITE_FUNCTION_ENDPOINT"] == null
+        || req.variables["APPWRITE_FUNCTION_API_KEY"] == null) {
+
+        return res.json(
+            data = mapOf(
+                "success" to false,
+                "message" to "Environment variables are not set, Function cannot use Appwrite SDK"
+            ),
+            statusCode = HTTP_BAD_REQEST
+        )
+    }
+
+    
+
+    // Check for databaseId and collectionId in the payload
+    val payloadMap = Gson().fromJson<Map<String, String>>(
+        req.payload.ifBlank { "{}" },
+        Map::class.java)
+    val databaseId = payloadMap["databaseId"] ?: ""
+    val collectionId = payloadMap["collectionId"] ?: ""
+    
+    if (databaseId.isEmpty()) {
+        return res.json(
+            data = mapOf(
+                "success" to false,
+                "message" to "Payload doesn't contain 'databaseId'"
+            ),
+            statusCode = HTTP_BAD_REQEST
+        )
+    }
+
+    if (collectionId.isEmpty()) {
+        return res.json(
+            data = mapOf(
+                "success" to false,
+                "message" to "Payload doesn't contain 'collectionId'"
+            ),
+            statusCode = HTTP_BAD_REQEST
+        )
+    }
+
+    val client = Client()
+        .setEndpoint(req.variables["APPWRITE_FUNCTION_ENDPOINT"]!!)
+        .setProject(req.variables["APPWRITE_FUNCTION_PROJECT_ID"]!!)
+        .setKey(req.variables["APPWRITE_FUNCTION_API_KEY"]!!)
+        .setSelfSigned(true)
+
+    var done = false
+    val databases = Databases(client)    
+
+    var document_list: DocumentList? = null; 
+    try {
+        document_list = databases.listDocuments(
```

Comment:
By default, only a maximum of 25 documents are returned by this API call. Because of this, if there are more than 25 documents, not all documents will be wiped.